explorer/src/components/instruction/anchor/types.ts

```diff
@@ -1,16 +1,21 @@
+import { Connection } from "@solana/web3.js";
 import { TransactionInstruction } from "@solana/web3.js";
-
-// list of programs written in anchor
-// - should have idl on-chain for GenericAnchorDetailsCard to work out of the box
-// - before adding another program to this list, please make sure that the ix
-// are decoding without any errors
-const knownAnchorPrograms = [
-  // https://github.com/blockworks-foundation/voter-stake-registry
-  "4Q6WW2ouZ6V3iaNm56MTd5n2tnTm4C5fiH8miFHnAFHo",
-];
+import { Program, Provider } from "@project-serum/anchor";
 
 export const isInstructionFromAnAnchorProgram = (
-  instruction: TransactionInstruction
+  url: string,
+  ix: TransactionInstruction
 ) => {
-  return knownAnchorPrograms.includes(instruction.programId.toBase58());
+  const checkIdl = async () => {
```

Comment:
This seems slow and done in the wrong spot. How about checking each instruction program id in the provider after we fetch the transaction? It would be nice to cache the results as well

Reply:
Added anchor hook to load in the Program based upon the programId, which uses a cache & avoids the memory leak in the `useEffect`